pontoon/base/middleware.py

```diff
@@ -83,3 +77,54 @@ def __call__(self, request):
 
         request.session["next_path"] = request.get_full_path()
         return redirect(email_consent_url)
+
+
+class ThrottleIpMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.max_count = settings.THROTTLE_MAX_COUNT
+        self.observation_period = settings.THROTTLE_OBSERVATION_PERIOD
+        self.block_duration = settings.THROTTLE_BLOCK_DURATION
+
+        # Set to block_duration if longer, otherwise requests will be blocked until
+        # the observation_period expires rather than when the block_duration expires
+        if self.observation_period > self.block_duration:
+            self.observation_period = self.block_duration
+
+    def __call__(self, request):
+        if settings.THROTTLE_ENABLED is False:
+            return self.get_response(request)
+
+        ip = get_ip(request)
+
+        # Generate cache keys
+        throttle_key = f"throttle_ip_{ip}"
+        blocked_key = f"blocked_ip_{ip}"
+
+        # Check if IP is currently blocked
+        if cache.get(blocked_key):
+            return JsonResponse({"detail": "Too Many Requests"}, status=429)
+
+        # Fetch current request count and timestamp
+        request_data = cache.get(throttle_key)
+        now = time.time()
+
+        if request_data:
+            request_count, first_request_time = request_data
+            if request_count >= self.max_count:
+                # Block further requests for block_duration seconds
+                cache.set(blocked_key, True, self.block_duration)
```

Comment:
Do you think it would be worth to log this action, to get a sense of how often we block IPs?